aptos-move/block-executor/src/types.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright © Aptos Foundation
+
+use aptos_types::transaction::BlockExecutableTransaction as Transaction;
+use std::{collections::HashSet, fmt};
+
+#[derive(Eq, Hash, PartialEq, Debug)]
+pub enum InputOutputKey<K, T, I> {
+    Resource(K),
+    Group(K, T),
+    DelayedField(I),
+}
+
+pub struct ReadWriteSummary<T: Transaction> {
+    pub reads: HashSet<InputOutputKey<T::Key, T::Tag, T::Identifier>>,
+    pub writes: HashSet<InputOutputKey<T::Key, T::Tag, T::Identifier>>,
+}
+
+impl<T: Transaction> ReadWriteSummary<T> {
+    pub fn conflicts_with_previous(&self, previous: &Self) -> bool {
```

Comment:
Maybe we can add txn index and assert self has a higher one than prev?

Reply:
does it make sense to have a super basic unit test here? i'm not sure how the implementation can change accidentally, but it should be easy to test the desired behavior.